### PR TITLE
[WeOpen-Star #3805] Add convenience annotation for `shenyu-client-springcloud` and `shenyu-client-springmvc`.

### PR DESCRIPTION
**Pull request by ShawnJim**
About #3805 .

---

### Task
- [X] add @ShenyuRequestMapping, @ShenyuGetMapping, @ShenyuPostMapping, @ShenyuPutMapping, @ShenyuDeleteMapping, @ShenyuPatchMapping for shenyu-client-springmvc; 
- [X] add @ShenyuRequestMapping, @ShenyuGetMapping, @ShenyuPostMapping, @ShenyuPutMapping, @ShenyuDeleteMapping, @ShenyuPatchMapping for shenyu-client-springcloud; 

---

Make sure that:

- [X] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [X] You submit test cases (unit or integration tests) that back your changes.
- [X] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-client/shenyu-client-http/shenyu-client-springcloud/src/main/java/org/apache/shenyu/client/springcloud/annotation/ShenyuRequestMapping.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    /**
+     * Path string.
+     *
+     * @return the string
+     */
+    @AliasFor(
+            attribute = "path"
+    )
+    String value() default "";
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-http/shenyu-client-springcloud/src/main/java/org/apache/shenyu/client/springcloud/annotation/ShenyuRequestMapping.java:46]: here should be
```java
    @AliasFor(
            annotation = RequestMapping.class
    )
```

Comment (dragon-zhang) [reply shenyu-client/shenyu-client-http/shenyu-client-springcloud/src/main/java/org/apache/shenyu/client/springcloud/annotation/ShenyuRequestMapping.java:46]: And it seems that you lost other methods in `@RequestMapping`.

Comment (ShawnJim) [reply shenyu-client/shenyu-client-http/shenyu-client-springcloud/src/main/java/org/apache/shenyu/client/springcloud/annotation/ShenyuRequestMapping.java:46]: get, i`ll fix for later. thanks for you review.

File: shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/test/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientEventListenerTest.java [hunk 2]
```diff
@@ -183,4 +193,33 @@ public String test(final String hello) {
...
+    @RestController
+    @ShenyuRequestMapping("/order")
+    static class SpringMvcClientTestBean5 {
+
+        @ShenyuPostMapping(value = "/save")
+        public String save(@RequestBody final String body) {
+            return "" + body;
+        }
+
+        @ShenyuGetMapping("/{id}")
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/test/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientEventListenerTest.java:205]: You need to modify the `shenyu-admin` module to adapt the writing of `${id}`, use regular expression to replace `/aa/${xxx}/cc` to `/aa/**/cc`

Comment (ShawnJim) [reply shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/test/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientEventListenerTest.java]: hi @dragon-zhang , Is that for rule matching?  i`ll modify and test it tomorrow.

Comment (dragon-zhang) [reply shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/test/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientEventListenerTest.java]: > hi @dragon-zhang , Is that for rule matching? i`ll modify and test it tomorrow.

yes

File: shenyu-client/shenyu-client-http/shenyu-client-springcloud/src/main/java/org/apache/shenyu/client/springcloud/annotation/ShenyuDeleteMapping.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+@Retention(RetentionPolicy.RUNTIME)
+@ShenyuRequestMapping(method = {RequestMethod.DELETE})
+public @interface ShenyuDeleteMapping {
+
+    /**
+     * Alias for {@link ShenyuRequestMapping#value}.
+     *
+     * @return the string
+     */
+    @AliasFor(attribute = "path")
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-http/shenyu-client-springcloud/src/main/java/org/apache/shenyu/client/springcloud/annotation/ShenyuDeleteMapping.java:42]: change to `@AliasFor(annotation = ShenyuRequestMapping.class)`

File: shenyu-client/shenyu-client-http/shenyu-client-springcloud/src/main/java/org/apache/shenyu/client/springcloud/annotation/ShenyuDeleteMapping.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    String value() default "";
+
+    /**
+     * Alias for {@link ShenyuRequestMapping#path}.
+     *
+     * @return the string
+     */
+    @AliasFor(attribute = "value")
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-http/shenyu-client-springcloud/src/main/java/org/apache/shenyu/client/springcloud/annotation/ShenyuDeleteMapping.java:50]: @AliasFor(annotation = ShenyuRequestMapping.class)

File: shenyu-client/shenyu-client-http/shenyu-client-springcloud/src/main/java/org/apache/shenyu/client/springcloud/annotation/ShenyuGetMapping.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+@Retention(RetentionPolicy.RUNTIME)
+@ShenyuRequestMapping(method = {RequestMethod.GET})
+public @interface ShenyuGetMapping {
+
+    /**
+     * Alias for {@link ShenyuRequestMapping#value}.
+     *
+     * @return the string
+     */
+    @AliasFor(attribute = "path")
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-http/shenyu-client-springcloud/src/main/java/org/apache/shenyu/client/springcloud/annotation/ShenyuGetMapping.java:42]: change to `@AliasFor(annotation = ShenyuRequestMapping.class)`

File: shenyu-client/shenyu-client-http/shenyu-client-springcloud/src/main/java/org/apache/shenyu/client/springcloud/annotation/ShenyuGetMapping.java [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    String value() default "";
+
+    /**
+     * Alias for {@link ShenyuRequestMapping#path}.
+     *
+     * @return the string
+     */
+    @AliasFor(attribute = "value")
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-http/shenyu-client-springcloud/src/main/java/org/apache/shenyu/client/springcloud/annotation/ShenyuGetMapping.java:50]: change to `@AliasFor(annotation = ShenyuRequestMapping.class)`

File: shenyu-client/shenyu-client-http/shenyu-client-springcloud/src/main/java/org/apache/shenyu/client/springcloud/annotation/ShenyuDeleteMapping.java (hunk 3 above)
Comment (ShawnJim) [reply shenyu-client/shenyu-client-http/shenyu-client-springcloud/src/main/java/org/apache/shenyu/client/springcloud/annotation/ShenyuDeleteMapping.java]: done

File: shenyu-client/shenyu-client-http/shenyu-client-springcloud/src/main/java/org/apache/shenyu/client/springcloud/annotation/ShenyuRequestMapping.java [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+     */
+    @AliasFor(annotation = RequestMapping.class)
+    String[] produces() default {};
+
+    /**
+     * Alias for {@link RequestMapping#method}.
+     *
+     * @return the RequestMethod[]
+     */
+    RequestMethod[] method() default {};
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-http/shenyu-client-springcloud/src/main/java/org/apache/shenyu/client/springcloud/annotation/ShenyuRequestMapping.java:116]: does here lost `@AliasFor(annotation = RequestMapping.class)` ?

File: shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/annotation/ShenyuRequestMapping.java [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+     */
+    @AliasFor(annotation = RequestMapping.class)
+    String[] produces() default {};
+
+    /**
+     * Alias for {@link RequestMapping#method}.
+     *
+     * @return the RequestMethod[]
+     */
+    RequestMethod[] method() default {};
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/annotation/ShenyuRequestMapping.java:125]: does here lost `@AliasFor(annotation = RequestMapping.class)` ?

Comment (ShawnJim) [reply shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/annotation/ShenyuRequestMapping.java:126]: done

File: shenyu-client/shenyu-client-http/shenyu-client-springcloud/src/main/java/org/apache/shenyu/client/springcloud/annotation/ShenyuRequestMapping.java (hunk 7 above)
Comment (ShawnJim) [reply shenyu-client/shenyu-client-http/shenyu-client-springcloud/src/main/java/org/apache/shenyu/client/springcloud/annotation/ShenyuRequestMapping.java:117]: done

**Review by dragon-zhang — APPROVED** [review]

**Pull request merged**